### DISCUSSION TITLE
Canvas with post auto translation in many languages

### DISCUSSION THREAD

**Discussion by dgillier** [austintoddj/canvas#1131] [General]
Hello,

I'm considering using Canvas for a multilingual blog.
By multilingual, I mean authors will wrote post in their own language and using Deepl a translate in 6/7 other languages will be done automatically...
Of course I want to do it once, not on every reading request ;)

Do you think I can do something like that ?
Is canvas compatible with any Multilingual model package (like spatie's one, or others) ?
If so I'm looking to have a listener triggered on some canvas event, and doing the deepl translation there, using the multilingual package to save the different translations...

Any thought about that ?

Thanks,
Denis

**Comment by austintoddj** [comment]
Hi @dgillier. Right now the extent of Canvas' multi-lingual support is restricted to the UI of an authenticated user. If users did author content in their own language, could you simply pass it through the DeepL API with a [package like this](https://github.com/DeepLcom/deepl-php) and present it in as many languages as you want?
